### PR TITLE
[CI/CD]Add XPU pipeline directory

### PR DESCRIPTION
**Pull request by VincyZhang**
**What this PR does / why we need it**:

Add the XPU Buildkite flow under `.buildkite/k3_tests/xpu/`.

This change introduces four files:
- `.buildkite/k3_tests/xpu/BK_WEB_SETUP.md`: Buildkite UI setup instructions and trigger rules for the XPU pipeline.
- `.buildkite/k3_tests/xpu/buildkite-pipeline.yml`: the Steps-editor snippet that uploads the real pipeline.
- `.buildkite/k3_tests/xpu/pipeline.yml`: the actual XPU smoke-test pipeline, using the public XPU image.
- `.buildkite/k3_tests/xpu/run.sh`: the XPU smoke-test entrypoint, which installs LMCache from source via `setup-lmcache-only-env.sh` and runs the smoke tests.

**Special notes for your reviewers**:

The XPU flow no longer builds or pushes a local image. It pulls the prebuilt public XPU image from ECR Public and installs LMCache from source inside the job pod.

**If applicable**:

- [x] this PR contains user facing changes - docs added
- [ ] this PR contains unit tests

### CODE REVIEW

File: .buildkite/k3_tests/xpu/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    command: bash "${BUILDKITE_BUILD_CHECKOUT_PATH:-.}/.buildkite/k3_tests/xpu/run.sh"
+    timeout_in_minutes: 180
+    agents: { queue: "xpu" }
+    plugins:
+      - kubernetes:
+          podSpec:
+            # Use the prebuilt public XPU CI image.
+            containers:
+              - name: container-0
+                image: public.ecr.aws/q9t5s3a7/vllm-ci-test-repo:ee0da84ab9e04ac7610e28580af62c365e898389-xpu
```
Comment (maobaolong) [inline .buildkite/k3_tests/xpu/pipeline.yml:15]: @ApostaC This looks like a fixed vLLM version. Do we need to use the latest verified version vLLM?

Ref to https://github.com/LMCache/LMCache/blob/buildkite_latest_tested_vllm/latest_tested_vllm.txt , but as this is the first PR of you, i'm also ok if you use a fixed vllm version first and then try to use the latest_tested_vllm as followup PR.

Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/pipeline.yml:15]: We can change to nightly build version in the future.

File: .buildkite/k3_tests/xpu/run.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+# XPU smoke test entrypoint.
+# Runs directly inside the agent-stack-k8s job pod.
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+REPO_ROOT="$(cd "${SCRIPT_DIR}/../../.." && pwd)"
+
+log() {
+  echo "[xpu-test] $*"
```
Comment (hlin99) [inline .buildkite/k3_tests/xpu/run.sh]: if this category is for unit test, can we rename to unit-tests-xpu, which is aligned with AMD naming

File: .buildkite/k3_tests/xpu/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    timeout_in_minutes: 180
+    agents: { queue: "intel-xpu" }
+    plugins:
+      - kubernetes:
+          podSpec:
+            # Use the prebuilt public XPU CI image.
+            containers:
+              - name: container-0
+                image: public.ecr.aws/q9t5s3a7/vllm-ci-test-repo:ee0da84ab9e04ac7610e28580af62c365e898389-xpu
+                imagePullPolicy: Never
```
Comment (hlin99) [inline .buildkite/k3_tests/xpu/pipeline.yml]: is flag IfNotPresent what we expect?

File: .buildkite/k3_tests/xpu/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# XPU smoke test.
+# Runs on a dedicated `xpu` queue using the public XPU CI image.
```
Comment (hlin99) [inline .buildkite/k3_tests/xpu/pipeline.yml]: ”intel-xpu“ ？align with line#8

File: .buildkite/k3_tests/xpu/run.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  echo "[xpu-test] ERROR: $*" >&2
+  exit 1
+}
+
+export TEST_SELECTOR="${TEST_SELECTOR:-calculate_cdf or get_gpu_pci_bus_id or load_and_reshape_flash}"
+cd "${REPO_ROOT}"
+
+source "${REPO_ROOT}/.buildkite/k3_harness/setup-lmcache-only-env.sh"
+
+if [ -f /opt/intel/oneapi/setvars.sh ]; then
```
Comment (hlin99) [inline .buildkite/k3_tests/xpu/run.sh:18]: shall we move this env configs to the top?

File: .buildkite/k3_tests/xpu/pipeline.yml (hunk 1 above)
Comment (zxue2) [reply .buildkite/k3_tests/xpu/pipeline.yml:15]: @ApostaC   is there any docker hub for LMCache pre-built docker image?

File: .buildkite/k3_tests/xpu/pipeline.yml (hunk 3 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/pipeline.yml]: Good catch!  `IfNotPresent` is more suitable.

File: .buildkite/k3_tests/xpu/pipeline.yml (hunk 4 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/pipeline.yml]: Yes, corrected the name.

File: .buildkite/k3_tests/xpu/run.sh (hunk 5 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/run.sh:18]: Will the position has influence in installation or running?

File: .buildkite/k3_tests/xpu/run.sh (hunk 2 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/run.sh]: Good point. I changed the log print and labels in scripts. But the displayed name in PR is set in Buildkite setting. 
We need help from maintainers to change the name.

Comment (hlin99) [reply .buildkite/k3_tests/xpu/run.sh]: okay. Got it. This naming is not controlled by this PR
Thx for confirmation

File: .buildkite/k3_tests/xpu/run.sh (hunk 5 above)
Comment (hlin99) [reply .buildkite/k3_tests/xpu/run.sh:18]: Not sure….but usually we setup env as first step

**Review by hlin99 — APPROVED** [review]
LGTM
Thanks @VincyZhang for setting xpu CI up. 
It won’t be a single step to make it ready. If we can have an issue listing rough plan moving forward. It will be perfect.

File: .buildkite/k3_tests/xpu/run.sh (hunk 5 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/run.sh:18]: It's okay, I moved it to the top.

File: .buildkite/k3_tests/xpu/run.sh (hunk 2 above)
Comment (hlin99) [reply .buildkite/k3_tests/xpu/run.sh]: PS: I’ve done the renaming to get aligned with amd unit test naming

**Review by maobaolong — APPROVED** [review]
@VincyZhang Thanks for your contribution. LGTM

File: .buildkite/k3_tests/xpu/run.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+source "${REPO_ROOT}/.buildkite/k3_harness/setup-lmcache-only-env.sh"
+
+log "installing job dependencies"
+uv pip install -r requirements/common.txt -r requirements/test.txt
+
+log "checking torch.xpu availability"
+python - <<'PY'
+import torch
+
+assert hasattr(torch, "xpu") and torch.xpu.is_available(), "Intel XPU not available in pod"
```
Comment (zxue2) [inline .buildkite/k3_tests/xpu/run.sh:36]: @VincyZhang , pls kindly check if the assertion can be placed at the top for fast-fail?

File: .buildkite/k3_tests/xpu/run.sh (hunk 5 above)
Comment (zxue2) [reply .buildkite/k3_tests/xpu/run.sh:18]: @VincyZhang , pls kindly add an assertion or else for fast-fail  in case /opt/intel/oneapi/setvars.sh not detected.     We need to ensure oneapi env source occurs prior to setup-lmcache-only-env.sh

**Comment by zxue2** [comment]
Given not all test cases under  ```tests/v1/```   ```tests/ ```  ```tests/benchmarks```, etc.  are applicable to xpu ci  immediately, intel team plans to filter the proper scope for now and may extend the scope later when all xpu features completed.  Let's have a meeting to consolidate it. @hlin99 @VincyZhang @ftian1 @XinyuYe-Intel 

CC: @maobaolong @ApostaC

File: .buildkite/k3_tests/xpu/run.sh (hunk 6 above)
Comment (VincyZhang) [reply .buildkite/k3_tests/xpu/run.sh:36]: I can manage it in a following PR. 
For now, it's not really necessary as long as we're using vLLM pre-built images.

Comment (zxue2) [reply .buildkite/k3_tests/xpu/run.sh:36]: torch.xpu.is_available  is h/w detection, irrelevant to pre-built image ...  the logic is not moving forward without xpu h/w .
